scodec/src/main/scala/io/github/finagle/serial/scodec/ScodecSerial.scala

```diff
@@ -1,34 +1,41 @@
 package io.github.finagle.serial.scodec
 
-import _root_.scodec.bits.BitVector
 import _root_.scodec.{Codec, Err}
+import _root_.scodec.bits.BitVector
 import _root_.scodec.codecs._
 import com.twitter.util.{Return, Throw, Try}
 import io.github.finagle.Serial
-import io.github.finagle.serial.{EncodingError, UnencodedError}
+import io.github.finagle.serial.{CodecError, ApplicationError}
 
 trait ScodecSerial extends Serial {
   type C[A] = Codec[A]
 
-  private[this] val stringWithLength: Codec[String] = variableSizeBits(uint24, utf8)
+  /**
+   * A codec for error message strings.
+   *
+   * Note that messages longer than 65536 characters are not supported.
+   */
+  private[this] lazy val stringWithLength: Codec[String] = variableSizeBits(uint24, utf8)
+
+  lazy val codecErrorCodec: Codec[CodecError] = stringWithLength.hlist.as[CodecError]
 
-  val encodingErrorCodec: Codec[EncodingError] = stringWithLength.hlist.as[EncodingError]
-  val unencodedErrorCodec: Codec[UnencodedError] = stringWithLength.hlist.as[UnencodedError]
+  lazy val unhandledApplicationErrorCodec: Codec[ApplicationError] =
+    stringWithLength.hlist.as[ApplicationError]
 
-  val domainErrorCodec: Codec[Throwable] = discriminated[Throwable].by(uint8)
-    .| (0) {
-      case ex: NumberFormatException => ex.getMessage
-    } (new NumberFormatException(_)) (stringWithLength)
+  /**
+   * By default we do not encode any application errors.
+   */
+  lazy val applicationErrorCodec: Codec[Throwable] = ApplicationErrorCodec.empty.underlying
```

Comment:
You can create a custom instance of `ScodecSerial` that overrides this value (see `ScodecApplicationErrorIntegrationTest` for an example). Does that seem reasonable?